### ISSUE TITLE
Same enum name in different namespace

### ISSUE THREAD

**Issue by cmouli84** [actgardner/gogen-avro#169]
Hi

We are running into an issue due to same enum name present in different namespaces. Find below the sample schema to illustrate the issue we are running into

```
{
  "type": "record",
  "name": "SomeType",
  "namespace": "com.company.team",
  "fields": [
    {
      "name": "fieldOne",
      "type": {
        "name": "TypeOne",
        "type": "record",
        "namespace": "com.company.team",
        "fields": [
          {
            "name": "type",
            "type": {
              "type": "enum",
              "name": "SomeEnum",
              "symbols": [
                "A",
                "B",
                "C"
              ]
            }
          }
        ]
      }
    },
    {
      "name": "fieldTwo",
      "type": {
        "name": "TypeTwo",
        "type": "record",
        "namespace": "com.company.shared",
        "fields": [
          {
            "name": "type",
            "type": {
              "type": "enum",
              "name": "SomeEnum",
              "symbols": [
                "X",
                "Y"
              ]
            }
          }
        ]
      }
    }
  ]
}
```

In the above schema, `SomeEnum` is present under namespaces `com.company.team` and `com.company.shared` and the enum in those namespaces allows for different set of values. But the generated go structs refer to same enum from `com.company.shared`.

```
type ComCompanyTeamSomeType struct {
	FieldOne ComCompanyTeamTypeOne `json:"fieldOne"`

	FieldTwo ComCompanySharedTypeTwo `json:"fieldTwo"`
}


type ComCompanyTeamTypeOne struct {
	Type SomeEnum `json:"type"`
}

type ComCompanySharedTypeTwo struct {
	Type SomeEnum `json:"type"`
}

type SomeEnum int32

const (
	SomeEnumX SomeEnum = 0
	SomeEnumY SomeEnum = 1
)
```

Command used to generate the go struct
```
gogen-avro -namespaced-names=full -package=schema schema/. schema.avsc
```

Is it possible to generate namespaced enums similar to go struct when `-namespaced-names=full` is set?

Thanks

**Comment by dylanmei** [comment]
The issue appears to be related: https://github.com/actgardner/gogen-avro/issues/13#issuecomment-278990580

**Comment by cmouli84** [comment]
Thanks @actgardner Issue is resolved with latest release (10.1.0)
